sycl/plugins/esimd_cpu/pi_esimd_cpu.cpp

```diff
@@ -962,16 +1541,36 @@ pi_result piEnqueueMemBufferRead(pi_queue Queue, pi_mem Src,
                                  pi_uint32 NumEventsInWaitList,
                                  const pi_event *EventWaitList,
                                  pi_event *Event) {
-  (void)Queue;
-  (void)Src;
-  (void)BlockingRead;
-  (void)Offset;
-  (void)Size;
-  (void)Dst;
-  (void)NumEventsInWaitList;
-  (void)EventWaitList;
-  (void)Event;
-  DIE_NO_IMPLEMENTATION;
+  if (BlockingRead) {
+    /// TODO : Support Blocked read, 'Queue' handling
```

Comment:
NIT. Suggest adding an assert with this message.

Reply:
Will apply.

Reply:
`assert()` or returning error code like `return PI_INVALID_OPERATION`? In `pi_level_zero.cpp`, `PI_ASSERT()` returns error code given as argument for the macro.

Reply:
Since blocking read operation is legal in PI and it is just a temporary limitation of esimd_cpu plugin, suggest putting an assert here, so we can quickly catch that a test fails due to still unimplemented functionality.

Reply:
@romanovvlad , @kbobrovs ,

What is recommended handling for temporary limitation? It looks there are three types of them - `assert()`, `throw sycl::(ERROR)`, and `return PI_INVALID_`.

> kbobrovs 8 days ago Contributor
> all temporary limitations should be RT exceptions (throw sycl::exception...) rather than asserts. asserts are not triggered in release builds, users will silently get wrong results.

Reply:
`assert()` is applied for BlockingRead support check.